theseus/geometry/tests/test_se2.py

```diff
@@ -34,55 +34,72 @@ def create_random_se2(batch_size, rng, dtype=torch.float64):
 
 
 @pytest.mark.parametrize("dtype", [torch.float32, torch.float64])
-def test_exp_map(dtype):
+@pytest.mark.parametrize("enable_functorch", [True, False])
+def test_exp_map(dtype, enable_functorch):
     rng = torch.Generator()
     rng.manual_seed(0)
+    ATOL = 2e-4 if dtype == torch.float32 else 1e-6
+
     for batch_size in [1, 20, 100]:
         theta = torch.from_numpy(np.linspace(-np.pi, np.pi, batch_size))
         u = torch.randn(batch_size, 2, dtype=dtype, generator=rng)
         tangent_vector = torch.cat([u, theta.unsqueeze(1)], dim=1)
-        check_exp_map(tangent_vector.to(dtype=dtype), th.SE2)
+        check_exp_map(
+            tangent_vector.to(dtype=dtype), th.SE2, enable_functorch=enable_functorch
+        )
+        check_projection_for_exp_map(
+            tangent_vector.to(dtype=dtype),
+            th.SE2,
+            atol=ATOL,
+            enable_functorch=enable_functorch,
+        )
 
 
 @pytest.mark.parametrize("dtype", [torch.float32, torch.float64])
-def test_log_map(dtype):
+@pytest.mark.parametrize("enable_functorch", [True, False])
+def test_log_map(dtype, enable_functorch):
```

Comment:
I am wondering what the usage looks like from the user side. Does the user pass `enable_functorch` to a `AutoDiffCostFunction` or is the flag internally set when you pass `autograd_functorch = True`.

We should also add the end-to-end or at least the `AutoDiffCostFunction` unit test for when using `autograd_functorch` and also `autograd_loop_over_batch` if we haven't already.

Finally I think it would be useful setup a benchmark script `/evaluations` to quickly run a comparison of the variant with dense, loop over batch, and functorch, so that we can evaluate future updates to our code or functorch more easily.

Reply:
This is internally set. I'm not sure if we still another test for AutoDiffCostFunction with `autograd_functorch=True` since these tests in Lie groups have already covered this. 

There are tests for `autograd_loop_over_batch=True`.